### ISSUE TITLE
MDDataTable - Sorting error | Missing documentation

### ISSUE THREAD

**Issue by podraco** [kivymd/KivyMD#919]
### Description of the Bug

This error shows when running the docs example
I think that the sorting method doesn't recognize the icons format inside raw_data
but i'm not sure if this is a common sorting error or, just the docs that are not updated properly, ill take a look on this

### Code and Logs

```python
from kivy.metrics import dp
from kivy.uix.anchorlayout import AnchorLayout

from kivymd.app import MDApp
from kivymd.uix.datatables import MDDataTable


class Example(MDApp):
    def build(self):
        layout = AnchorLayout()
        self.data_tables = MDDataTable(
            size_hint=(0.7, 0.6),
            use_pagination=True,
            check=True,
            column_data=[
                ("No.", dp(30)),
                ("Status", dp(30)),
                ("Signal Name", dp(60), self.sort_on_signal),
                ("Severity", dp(30)),
                ("Stage", dp(30)),
                ("Schedule", dp(30), self.sort_on_schedule),
                ("Team Lead", dp(30), self.sort_on_team)
            ],
            row_data=[
                ("1", ("alert", [255 / 256, 165 / 256, 0, 1], "No Signal"),
                 "Astrid: NE shared managed", "Medium", "Triaged", "0:33",
                 "Chase Nguyen"),
                ("2", ("alert-circle", [1, 0, 0, 1], "Offline"),
                 "Cosmo: prod shared ares", "Huge", "Triaged", "0:39",
                 "Brie Furman"),
                ("3", (
                    "checkbox-marked-circle",
                    [39 / 256, 174 / 256, 96 / 256, 1],
                    "Online"), "Phoenix: prod shared lyra-lists", "Minor",
                 "Not Triaged", "3:12", "Jeremy lake"),
                ("4", (
                    "checkbox-marked-circle",
                    [39 / 256, 174 / 256, 96 / 256, 1],
                    "Online"), "Sirius: NW prod shared locations",
                 "Negligible",
                 "Triaged", "13:18", "Angelica Howards"),
                ("5", (
                    "checkbox-marked-circle",
                    [39 / 256, 174 / 256, 96 / 256, 1],
                    "Online"), "Sirius: prod independent account",
                 "Negligible",
                 "Triaged", "22:06", "Diane Okuma"),

            ],
            sorted_on="Schedule",
            sorted_order="ASC",
            elevation=2
        )
        self.data_tables.bind(on_row_press=self.on_row_press)
        self.data_tables.bind(on_check_press=self.on_check_press)
        layout.add_widget(self.data_tables)
        return layout

    def on_row_press(self, instance_table, instance_row):
        '''Called when a table row is clicked.'''

        print(instance_table, instance_row)

    def on_check_press(self, instance_table, current_row):
        '''Called when the check box in the table row is checked.'''

        print(instance_table, current_row)

    def sort_on_signal(self, data):
        return sorted(data, key=lambda l: l[2])

    def sort_on_schedule(self, data):
        return sorted(data, key=lambda l: sum([int(l[-2].split(":")[0])*60, int(l[-2].split(":")[1])]))

    def sort_on_team(self, data):
        return sorted(data, key=lambda l: l[-1])

Example().run()
"""


class MainApp(App):
    def build(self):
        self.root = Builder.load_string(kv)


if __name__ == '__main__':
    MainApp().run()
```

### Versions

* OS:  Ubuntu
* Python:  3.8
* Kivy: 2.0+
* KivyMD: Master

**Comment by podraco** [comment]
The code now requires that the sorting method returns 2 objects

the data that's sorted and indices
```py
    indices, sorted_data = self.sort_action(self.table_data.row_data)
```
What kind of indices is the class using now?

**Comment by podraco** [comment]
@kevin-schumann

**Comment by kevin-schumann** [comment]
@podraco The table is using both return values. Indices are used to restore the checks. In my pull request I've provided the following sorting function: 

```python
def sort_on_signal(self, data):
        ''' 
        Returns a
             - Tuple of indices (the permutation to sort the data in ascending order)
             - Tuple of the ascending ordered data
        '''
        return zip(*sorted(enumerate(data), key=lambda l: int(l[1][3])))
```

The function sort_on_team in your example can be converted to the new interface as follows:

```python
def sort_on_team(self, data):
        return zip(*sorted(enumerate(data), key=lambda l: l[1][-1]))
```

If wanted I can modify my code to be compatible with the older and/or provide documentation for the newer user-provided sort functions.

EDIT:
The indices indicate how the non-sorted data was arranged to sort it e.g how the rows needed to be moved.

**Comment by kevin-schumann** [comment]
@podraco To make it clearer and convenient, I've updated every sort-function provided in your code.

```python
def sort_on_signal(self, data):
    return zip(*sorted(enumerate(data), key=lambda l: l[1][2]))

def sort_on_schedule(self, data):
    return zip(*sorted(enumerate(data), key=lambda l: sum([int(l[1][-2].split(":")[0])*60, int(l[1][-2].split(":")[1])])))

def sort_on_team(self, data):
    return zip(*sorted(enumerate(data), key=lambda l: l[1][-1]))
```

**Comment by kevin-schumann** [comment]
@podraco I don't consider this bug per se, rather the current documentation doesn't reflect the currently needed function signature.

**Comment by podraco** [comment]
Thanks for your prompt response.

i'm actually working on a function that update the table contents and found this error when i was testing some code i used to sort and patch the class.

i'll add the documentation to my mods and refer this issue to it.

**Comment by podraco** [comment]
fixed by #921
